File: scanner/src/main/java/com/shipitdone/scanner/manager/ScannerManager.java
package com.shipitdone.scanner.manager;

import android.support.annotation.NonNull;
import android.view.KeyEvent;

public interface ScannerManager {
    void init();

    void recycle();

    void setScannerListener(@NonNull ScannerVariantManager.ScanListener listener);

    void sendKeyEvent(KeyEvent key);

    int getScannerModel();

    void scannerEnable(boolean enable);

    void setScanMode(String mode);

    void setDataTransferType(String type);

    void singleScan(boolean bool);

    void continuousScan(boolean bool);
}

File: scanner/src/main/java/com/shipitdone/scanner/manager/ScannerManager.kt
package com.shipitdone.scanner.manager

import android.view.KeyEvent
import android.content.Context
import com.shipitdone.scanner.manager.ScannerVariantManager.ScanListener

interface ScannerManager {
    fun init(context: Context)

    fun recycle(context: Context)

    fun setScannerListener(listener: ScanListener)

    fun sendKeyEvent(key: KeyEvent?)

    fun scannerEnable(context: Context, enable: Boolean)

    fun setScanMode(mode: String?)

    fun setDataTransferType(type: String?)

    fun singleScan(context: Context, bool: Boolean)

    fun continuousScan(context: Context, bool: Boolean)

    fun getScannerModel(): Int
}
